lib/liquid/static_registers.rb

```diff
@@ -25,8 +25,16 @@ def delete(key)
       @registers.delete(key)
     end
 
-    def fetch(key, default = nil)
-      key?(key) ? self[key] : default
+    UNDEFINED = Object.new
+
+    def fetch(key, default = UNDEFINED, &block)
+      if @registers.key?(key)
+        @registers.fetch(key)
+      elsif default != UNDEFINED
+        @static.fetch(key, default, &block)
```

Comment:
@tjoyal This block pass, and the one below is not covered by the tests you added in this PR.

Was the addition of the block pass intentional? If so: I suggest to add a test that shows this intention, else the block pass could be removed by a co-worker and break assumptions elsewhere, without tests of this unit noticing.

Reply:
You're right, the tests pass with

```diff
diff --git a/lib/liquid/static_registers.rb b/lib/liquid/static_registers.rb
index 0293f5f..94cb660 100644
--- a/lib/liquid/static_registers.rb
+++ b/lib/liquid/static_registers.rb
@@ -31,9 +31,9 @@ module Liquid
       if @registers.key?(key)
         @registers.fetch(key)
       elsif default != UNDEFINED
-        @static.fetch(key, default, &block)
+        @static.fetch(key, default)
       else
-        @static.fetch(key, &block)
+        @static.fetch(key)
       end
     end
 
```

> Was the addition of the block pass intentional?

Yes, the intention was to have it behave like a Hash#fetch, so it makes sense to pass along the block.

Reply:
k, cool. So I can build on this block pass behavior? Its intentionally part of the public API of this class (and will be in the future)?

Reply:
Yes.  We will also accept tests to help preserve that behaviour.